compiler/rustc_hir_analysis/src/collect/dump.rs

```diff
@@ -0,0 +1,43 @@
+use rustc_hir::def::DefKind;
+use rustc_hir::def_id::CRATE_DEF_ID;
+use rustc_middle::ty::TyCtxt;
+use rustc_span::sym;
+
+pub(crate) fn opaque_hidden_types(tcx: TyCtxt<'_>) {
+    if !tcx.has_attr(CRATE_DEF_ID, sym::rustc_hidden_type_of_opaques) {
+        return;
+    }
+
+    for id in tcx.hir().items() {
+        let DefKind::OpaqueTy = tcx.def_kind(id.owner_id) else { continue };
+
+        let ty = tcx.type_of(id.owner_id).instantiate_identity();
+
+        tcx.dcx().emit_err(crate::errors::TypeOf { span: tcx.def_span(id.owner_id), ty });
+    }
+}
+
+pub(crate) fn predicates_and_item_bounds(tcx: TyCtxt<'_>) {
+    for id in tcx.hir_crate_items(()).owners() {
```

Comment:
Instead of going through all owner items I could instead extend `CheckAttrVisitor::check_attributes`. That's how `#[rustc_object_lifetime_default]` is impl'ed.

Reply:
I don't think it really matters 🤷